pennylane/qchem/openfermion_obs.py

```diff
@@ -1013,6 +1014,16 @@ def _(
     load_data=False,
     convert_tol=1e12,
 ):
+
+    if unit.strip().lower() not in ("angstrom", "bohr"):
+        raise ValueError(
+            f"The provided unit, '{unit}' is not supported. "
+            f"Please set 'unit' to 'Bohr' or 'Angstrom'."
+        )
+
+    if unit.strip().lower() == "angstrom":
+        coordinates = coordinates / bohr_angs
```

Comment:
```suggestion
    if (coord_unit := unit.strip().lower()) not in ("angstrom", "bohr"):
        raise ValueError(
            f"The provided unit, '{unit}' is not supported. "
            f"Please set 'unit' to 'Bohr' or 'Angstrom'."
        )

    if coord_unit == "angstrom":
        coordinates = coordinates / bohr_angs
```
We might want to repeat the stripping.